### PR TITLE
chore: flowable users list (AR-2477)

### PR DESCRIPTION
**Pull request by Garzas**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues
When we are on the search users list they will not refresh after some user removal.

### Solutions
Make user list flowable to observe changes

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/GetAllContactsNotInConversationUseCase.kt [hunk 1]
```diff
@@ -5,14 +5,16 @@ import com.wire.kalium.logic.data.id.QualifiedID
...
 import com.wire.kalium.logic.data.user.UserRepository
 import com.wire.kalium.logic.functional.fold
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
 
 class GetAllContactsNotInConversationUseCase internal constructor(
     private val userRepository: UserRepository
 ) {
-    suspend operator fun invoke(conversationId: QualifiedID) =
+    suspend operator fun invoke(conversationId: QualifiedID): Flow<Result> =
```
Comment (yamilmedina) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/GetAllContactsNotInConversationUseCase.kt:14]:
```suggestion
    suspend operator fun invoke(conversationId: ConversationId): Flow<Result> =
```

Also, we take the opportunity to add some `kdocs` about this use case purpose :D  
What do you think ?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/publicuser/GetAllContactsUseCase.kt [hunk 2]
```diff
@@ -4,19 +4,20 @@ import com.wire.kalium.logic.StorageFailure
...
 
 internal class GetAllContactsUseCaseImpl internal constructor(
     private val userRepository: UserRepository
 ) : GetAllContactsUseCase {
 
-    override suspend fun invoke(): GetAllContactsResult =
-        userRepository.getAllKnownUsers()
-            .fold(GetAllContactsResult::Failure, GetAllContactsResult::Success)
+    override suspend fun invoke(): Flow<GetAllContactsResult> =
+        userRepository.observeAllKnownUsers()
```
Comment (yamilmedina) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/publicuser/GetAllContactsUseCase.kt:19]: We take the opportunity to add some kdocs about this use case purpose :D

File: persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConversationDAOImpl.kt [hunk 3]
```diff
@@ -445,8 +445,9 @@ class ConversationDAOImpl(
 
     override suspend fun revokeOneOnOneConversationsWithDeletedUser(userId: UserIDEntity) {
         conversationQueries.transaction {
-            val conversationId = memberQueries.selectConversationByMember(userId).executeAsOne().conversation
-            conversationQueries.updateConversationType(ConversationEntity.Type.GROUP, conversationId)
+            memberQueries.selectConversationByMember(userId).executeAsOneOrNull()?.conversation?.let { conversationId ->
```
Comment (yamilmedina) [inline persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConversationDAOImpl.kt:448]: Question here, is this change related to the scope of the task ? 🤔 Just to understand myself a bit better this how this affects

**Review by yamilmedina — COMMENTED** [review]
LGTM ! Just added a couple of suggestions 🥇

File: persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConversationDAOImpl.kt (hunk 3 above)
Comment (Garzas) [reply persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConversationDAOImpl.kt:448]: I saw an error that the user was not found in the logs in this function so I secured it

Comment (yamilmedina) [reply persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConversationDAOImpl.kt:448]: Perfect ! 👌

**Review by yamilmedina — APPROVED** [review]
Thanks for the additional docs ! 🎸
